.github/workflows/php-qa.yml

```diff
@@ -1,29 +1,60 @@
+            - '**.php'
+            - '**phpcs.xml.dist'
+            - '**psalm.xml'
+            - '**composer.json'
+    pull_request:
+        paths:
+            - '**workflows/php-qa.yml'
+            - '**.php'
+            - '**phpcs.xml.dist'
+            - '**psalm.xml'
+            - '**composer.json'
+    workflow_dispatch:
+        inputs:
+            jobs:
+                required: true
+                type: choice
+                default: 'Run all'
+                description: 'Choose jobs to run'
+                options:
+                    - 'Run all'
+                    - 'Run PHPCS only'
+                    - 'Run Psalm only'
+                    - 'Run lint only'
 
 concurrency:
-  group: ${{ github.workflow }}-${{ github.ref }}
-  cancel-in-progress: true
+    group: "${{ github.workflow }}-${{ github.ref }}"
+    cancel-in-progress: true
 
 jobs:
-  lint-php:
-    if: ${{ (github.event_name == 'workflow_dispatch') || (!contains(github.event.head_commit.message, 'skip lint')) }}
-    strategy:
-      matrix:
-        php: ["7.2", "7.3", "7.4", "8.0", "8.1", "8.2"]
-    uses: inpsyde/reusable-workflows/.github/workflows/lint-php.yml@main
-    with:
-      PHP_VERSION: ${{ matrix.php }}
+    lint:
+        if: ${{ (github.event_name != 'workflow_dispatch') || ((github.event.inputs.jobs == 'Run all') || (github.event.inputs.jobs == 'Run lint only')) }}
+        uses: inpsyde/reusable-workflows/.github/workflows/lint-php.yml@main
+        strategy:
+            matrix:
+                php: [ '7.4', '8.0', '8.1', '8.2', '8.3' ]
+        with:
+            PHP_VERSION: ${{ matrix.php }}
+            LINT_ARGS: '-e php --colors --show-deprecated ./src'
 
-  coding-standards-analysis-php:
-    if: ${{ (github.event_name == 'workflow_dispatch') || (!contains(github.event.head_commit.message, 'skip cs')) }}
-    needs: lint-php
-    uses: inpsyde/reusable-workflows/.github/workflows/coding-standards-php.yml@main
-    with:
-      PHPCS_ARGS: '--report=summary'
+    coding-standards-analysis:
+        if: ${{ (github.event_name != 'workflow_dispatch') || ((github.event.inputs.jobs == 'Run all') || (github.event.inputs.jobs == 'Run PHPCS only')) }}
+        uses: inpsyde/reusable-workflows/.github/workflows/coding-standards-php.yml@main
+        with:
+            PHP_VERSION: '8.3'
 
-  static-analysis-php:
-    if: ${{ (github.event_name == 'workflow_dispatch') || (!contains(github.event.head_commit.message, 'skip sa')) }}
-    needs: lint-php
-    uses: inpsyde/reusable-workflows/.github/workflows/static-analysis-php.yml@main
+    static-code-analysis:
+        if: ${{ (github.event_name != 'workflow_dispatch') || ((github.event.inputs.jobs == 'Run all') || (github.event.inputs.jobs == 'Run Psalm only')) }}
+        uses: inpsyde/reusable-workflows/.github/workflows/static-analysis-php.yml@main
+        strategy:
+            matrix:
+                php: [ '7.4', '8.0', '8.1', '8.2', '8.3' ]
```

Comment:
why do we need to do static analysis for different PHP versions? 👀

Reply:
Psalm has the option `--php-version` that runs analysis specific for that version.

We have it embedded in [our reusable workflow](https://github.com/inpsyde/reusable-workflows/blob/287807348eb92e3472809f5fd733edcdf34e7488/.github/workflows/static-analysis-php.yml#L66).

For example, some PHP functions accept a different parameter type depending on the PHP version. 

Some functions are deprecated, and some are added. If we don't pass any version, Psalm will consider the minimum version, so 7.4, and might find no errors. But then in more recent versions, we could have issues.

On the contrary, we might pass the last supported version (8.3 right now), and Psalm might find no issues, e.g. if we use a function that did not exist in 7.4, but then someone using 7.4 will have a fatal error.

Psalm has specific stubs [per PHP version](https://github.com/vimeo/psalm/tree/5.x/stubs) and depending on `--php-version` will use one stub or another.

So if you want the correct results for all supported versions you have to run the tool with all the versions you support.